### ISSUE TITLE
Volk build fails because of a mistake in CMake.txt

### ISSUE THREAD

**Issue by metablaster** [zeux/volk#212]
Target `CMake.txt` file where the issue is present:

```cmake
volk/CMakeLists.txt
```

Line number is 70 and reads `set(VOLK_INCLUDES "${VULKAN_HEADERS_INSTALL_DIR}/include")`</br>
This is incorrect because it should be set to `set(VOLK_INCLUDES "${VULKAN_HEADERS_INSTALL_DIR}")`

Reason why this is incorrect is because all `Khronos` and `LunarG` repositories do it this way (not specifying `/include` suffix)

What happens is that `vulkan.h` header is not found.

A workaround is to remove the `include` suffix but I propose that this be addressed by removing the `/include` subdirectory.

I'm willing to make a fork and pull request just let me know into which branch should I push it.

**Comment by metablaster** [comment]
You're correct and I apologize for possibly false positive, the mistake is likely on my part because of VSCode `VULKAN_HEADERS_INSTALL_DIR` CMake configuration variable which requires to specify `/include` suffix as expected for most Khronos repos, however this doesn't work for volk because it's already set in CMake.txt, so the issue can be be fixed on my part by adjusting CMake Tools settings in VSCode.

The only unexpected issue on volk's part is that `VULKAN_HEADERS_INSTALL_DIR` should be fully configurable by user instead of being hardcoded in CMake.txt.

Out of 15 Vulkan related repositories that's how it works, I sumbled upon the problem in volk only.

Bottom line is that `VULKAN_HEADERS_INSTALL_DIR` should not include any `/include` suffix but should instead be specified by library user.

**Comment by zeux** [comment]
> Bottom line is that VULKAN_HEADERS_INSTALL_DIR should not include any /include suffix but should instead be specified by library user.

Sorry, I still don't see why. Can you list the repositories you are using that use VULKAN_HEADERS_INSTALL_DIR and don't append `/include` to it in CMake setup?

**Comment by metablaster** [comment]
Here are examples I could quickly find showing no `/include` is set:

- https://github.com/LunarG/VulkanTools/blob/main/scripts/CMakeLists.txt (line 124)

```cmake
if (VULKAN_HEADERS_INSTALL_DIR)
    list(APPEND CMAKE_PREFIX_PATH ${VULKAN_HEADERS_INSTALL_DIR})
endif()
```
- https://github.com/KhronosGroup/Vulkan-Loader/blob/main/scripts/CMakeLists.txt (line 127)

```cmake
if (VULKAN_HEADERS_INSTALL_DIR)
    list(APPEND CMAKE_PREFIX_PATH ${VULKAN_HEADERS_INSTALL_DIR})
    set(CMAKE_REQUIRE_FIND_PACKAGE_VulkanHeaders TRUE PARENT_SCOPE)
endif()
```

- https://github.com/KhronosGroup/Vulkan-Profiles/blob/main/scripts/CMakeLists.txt (line 173)

```cmake
if (VULKAN_HEADERS_INSTALL_DIR)
    list(APPEND CMAKE_PREFIX_PATH ${VULKAN_HEADERS_INSTALL_DIR})
endif()
```

While all of these samples are related to `CMAKE_PREFIX_PATH` it's worth noting you could not find in those sample repositories any mention of `/include` leaf in any of the other CMake.txt files which suggests that `VULKAN_HEADERS_INSTALL_DIR` is used as it is set by user and not suffixed.

I'm not skilled with CMake but that's the best I could find by using search function in my build tree and didn't find any mention of `/include` anywhere else among over 10 repositories.

**Comment by zeux** [comment]
CMAKE_PREFIX_PATH is completely different from the include path setup.

To be clear, the way you are supposed to use VULKAN_HEADERS_INSTALL_DIR is that it should point to the installation location of the project https://github.com/KhronosGroup/Vulkan-Headers. That folder will contain `include` folders among others.

**Comment by metablaster** [comment]
It's a valid point that `VULKAN_HEADERS_INSTALL_DIR` should point to root directory instead of `/include` subdirectory.

However for what ever reason I have to specify `/include` manually for sample projects in my previous post, but must not specify it for volk because directory will result in `VULKAN_HEADERS_INSTALL_DIR/include/include` which is not right.

Not a big issue, I'll work on VSCode settings to workaround this somehow because currently I use same settings for all repos which is IMO not great.

Feel free to close this issue, it won't bring much attention because rarely who will build Vulkan from source, and likely not using VSCode.

**Comment by zeux** [comment]
Alright - after a little testing I'm very sure this is a configuration issue on your end. I tested Vulkan-Loader and volk and to build both, you need to specify `-D VULKAN_HEADERS_INSTALL_DIR=path-to-dir` without an /include suffix; the way you get that folder is by running the *installation* step on Vulkan-Headers project. Here's what the folder looks like for me:

```
~/build/vkhi $ ls -la
total 16
drwxrwxr-x 4 zeux zeux 4096 Nov  6 17:29 .
drwxrwxr-x 6 zeux zeux 4096 Nov  6 17:32 ..
drwxrwxr-x 4 zeux zeux 4096 Nov  6 17:29 include
drwxrwxr-x 4 zeux zeux 4096 Nov  6 17:29 share
~/build/vkhi $ ls include/
vk_video  vulkan
~/build/vkhi $ ls share/
cmake  vulkan
~/build/vkhi $ ls share/cmake/
VulkanHeaders
```

volk will only use the `include/` path; other projects you listed load modules from `share/cmake`, and for that to work you need to specify the path without `/include` as well.
